File: android/app/src/main/java/com/smartexpense/android/data/remote/dto/response/UserProfileResponseDto.java
package com.smartexpense.android.data.remote.dto.response;

public class UserProfileResponseDto {
    private String id;
    private String email;
    private String displayName;
    private String avatarUrl;
    private double monthlyBudget;
    private int streakDays;
    private String themeColor;

    public String getId() { return id; }
    public String getEmail() { return email; }
    public String getDisplayName() { return displayName; }
    public String getAvatarUrl() { return avatarUrl; }
    public double getMonthlyBudget() { return monthlyBudget; }
    public int getStreakDays() { return streakDays; }
    public String getThemeColor() { return themeColor; }
}

File: android/app/src/main/kotlin/com/smartexpense/android/data/remote/dto/response/UserProfileResponseDto.kt
package com.smartexpense.android.data.remote.dto.response

import com.google.gson.annotations.SerializedName

data class UserProfileResponseDto(
    @SerializedName("id") val id: String,
    @SerializedName("email") val email: String,
    @SerializedName("displayName") val displayName: String,
    @SerializedName("avatarUrl") val avatarUrl: String?,
    @SerializedName("monthlyBudget") val monthlyBudget: Double,
    @SerializedName("streakDays") val streakDays: Int,
    @SerializedName("themeColor") val themeColor: String?
)
